tests/Unit/NumericalAlgorithms/Interpolation/Test_LinearRegression.cpp

```diff
@@ -0,0 +1,66 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <array>
+#include <cmath>
+#include <cstddef>
+#include <random>
+#include <vector>
+
+#include "DataStructures/DataVector.hpp"
+#include "Framework/TestHelpers.hpp"
+#include "NumericalAlgorithms/Interpolation/LinearRegression.hpp"
+
+namespace {
+
+void test_linear_regression_datavector(const double expected_intercept,
+                                       const double expected_slope) {
+  const DataVector x_values{1.0, 2.0, 3.0, 4.0, 5.0};
+  const DataVector y_values = expected_slope * x_values + expected_intercept;
+  const auto result = intrp::linear_regression(x_values, y_values);
+
+  CHECK(result[0] == approx(expected_intercept));
+  CHECK(result[1] == approx(expected_slope));
+  // Error bars are zero because we have exactly a straight line.
+  CHECK(result[2] == approx(0.0));
+  CHECK(result[3] == approx(0.0));
```

Comment:
Test some case where these are not zero.

Reply:
Yes, I agree it would be nice to do this.  Does anyone have a suggestion of how to do such a test?  I can send in data points that are not exactly on a straight line, but I don't know how to predict the error bars, except to implement an independent linear regression solver.

I could simply test that the error bars are nonzero in that case, but that seems cheating.

Reply:
You don't need to do an RNG test.  Pick three points and work out the answer by hand.  (Four if there's an `N - 2` factor or something else that drops out in the N=3 case.)